src/docs/markdown/introduction.md

```diff
@@ -0,0 +1,276 @@
+
+Usually this means logging into your DNS provider and creating or changing the A (IPv4) and/or AAAA (IPv6) record for your domain (it can be a subdomain). We'll use `example.com` for this tutorial. Verify it has been set by running `dig example.com`.
+
+Then, **verify that your server's IP is publicly routable on the standard Web ports (80 and 443).** Ensure there are no firewalls or routers blocking these ports. On a home network, you may need to forward those ports to your machine (just be aware that your machine will become publicly accessible on those ports).
+
+Once your DNS and network infrastructure are properly configured, **all you need to do is replace `:80` in the Caddyfile with your domain name:**
+
+```caddy
+example.com {
+	root * /var/www/html
+	file_server
+}
+```
+
+Then reload the config once again:
+
+<pre><code class="cmd bash">sudo systemctl reload caddy</code></pre>
+
+Watch the logs to make sure it works:
+
+<pre><code class="cmd bash">journalctl -u caddy -f</code></pre>
+
+- **If it succeeds:** navigate to your site in your browser and see your site served over HTTPS, just like that!
+- **If it fails:** refer to the [troubleshooting tips](#troubleshooting) below.
+
+As you have just seen, Caddy serves sites over HTTPS automatically and by default (unless you explicitly configure `http://` or the HTTP port). As long as you keep your network and DNS properly configured, Caddy will keep your certificates renewed automatically.
+
+Caddy is the only server that works like this!
+
+<aside class="complete">Serve your site over HTTPS</aside>
+
+If you don't want to use a public domain name or are running this internally or locally instead, you can easily have Caddy use [fully-managed self-signed certificates](/docs/automatic-https#local-https) by specifying either:
+
+- your local/internal IP address,
+- the hostname `localhost`,
+- or any hostname that ends in `.localhost`
+
+instead of a registered public domain name.
+
+
+## Reverse proxy
+
+Oftentimes, your site consists of a backend application, but you want to put Caddy in front to handle TLS, routing, and other network-related details. Caddy's proxy is easy to use and extremely powerful.
+
+For example, if we have a backend that provides the site's API endpoints, we can easily proxy those with just 1 additional line of configuration:
+
+```caddy
+example.com {
+	root * /var/www/html
+	file_server
+	reverse_proxy /api/* 127.0.0.1:9000
+}
+```
+
+Notice the `reverse_proxy` directive. The first argument, `/api/*`, is a [path matcher](/docs/caddyfile/matchers) which filters only requests within `/api/`. Then it proxies those to the backend app listening on :9000.
+
+If your backend is a PHP app, simply replace the `reverse_proxy` directive with the `php_fastcgi` directive:
+
+```caddy
+example.com {
+	root * /var/www/html
+	file_server
+	php_fastcgi /api/* 127.0.0.1:9000
```

Comment:
I think this is misleading, because most of the time, this won't work.

For PHP apps, you need to configure `root` to your app's webroot, which should almost never be the actual root of the project, but actually a `public/` dir inside of the project. And if you use a path matcher, it doesn't change the `try_files` logic within the directive, so it will still look for `index.php` at the configured `root`, and not inside of some `/api` directory within the project.